### PR TITLE
BUGFIX: SD-546 Hide overlapped labels for Dual Axis with Stack

### PR DESCRIPTION
**Pull request by ptquang86**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist
- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)
- [x] Squash commits

# Related PRs
- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2374
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2152

# Related Jira tasks
- SD-546: https://jira.intgdc.com/browse/SD-546

### CODE REVIEW

**Comment by ptquang86** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4436/ : SUCCESS in 8m 35s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5687/ : SUCCESS in 12s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2384/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1775/ : UNSTABLE in 10m 57s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5688/ : SUCCESS in 9s

**Comment by ptquang86** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4447/ : SUCCESS in 10m 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5706/ : SUCCESS in 7s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2389/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1786/ : UNSTABLE in 14m 01s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5707/ : SUCCESS in 8s

**Comment by ptquang86** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4451/ : SUCCESS in 4m 33s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5715/ : SUCCESS in 7s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2391/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1792/ : SUCCESS in 11m 12s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5716/ : SUCCESS in 7s

**Comment by ptquang86** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4452/ : SUCCESS in 4m 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5717/ : SUCCESS in 7s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2392/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1793/ : UNSTABLE in 11m 00s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5718/ : SUCCESS in 12s

**Comment by ptquang86** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4453/ : SUCCESS in 7m 24s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5719/ : SUCCESS in 8s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2393/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1794/ : UNSTABLE in 13m 53s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5720/ : SUCCESS in 10s

**Comment by ptquang86** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4454/ : SUCCESS in 6m 25s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5721/ : SUCCESS in 12s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2394/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1795/ : SUCCESS in 11m 25s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5722/ : SUCCESS in 12s

**Comment by ptquang86** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1106.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1796/ : SUCCESS in 29m 56s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5723/ : SUCCESS in 7s

File: src/components/visualizations/chart/highcharts/plugins/autohideLabels/autohideColumnLabels.ts [hunk 1]
```diff
@@ -130,6 +133,45 @@ export function areNeighborsOverlapping(neighbors: IDataLabelsConfig[][]) {
...
+        const { x, y, width, height }: IClientRect = label.element.getBoundingClientRect();
+        const labelAttr: IRectBySize = {
+            x,
+            y,
+            width,
+            height,
+        };
+
+        return visiblePoints.some((point: IPointData) => {
+            const chartType: string = get(point, "series.options.type");
```
Comment (lanhq) [inline src/components/visualizations/chart/highcharts/plugins/autohideLabels/autohideColumnLabels.ts]: `chartType` => `seriesType`

Comment (ptquang86) [reply src/components/visualizations/chart/highcharts/plugins/autohideLabels/autohideColumnLabels.ts]: updated

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4470/ : SUCCESS in 5m 13s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5745/ : SUCCESS in 13s

**Comment by lanhq** [comment]
LGTM

File: src/components/visualizations/chart/highcharts/plugins/autohideLabels/autohideColumnLabels.ts [hunk 2]
```diff
@@ -176,7 +220,11 @@ function toggleStackedLabelsForDualAxis() {
         const neighbors = toNeighbors(labels);
         const areOverlapping = areNeighborsOverlapping(neighbors);
 
-        if (areOverlapping) {
+        const areOverlappingColumns = areOverlapping
```
Comment (pbenes) [inline src/components/visualizations/chart/highcharts/plugins/autohideLabels/autohideColumnLabels.ts]: I think we could have the naming more suggestive, and then use simple if condition with the two composed, something like (not tested): 
```
        const neighborsOverlapping = areNeighborsOverlapping(neighbors);
        const areOverlapping = neighborsOverlapping
            ? true
            : areLabelsOverlappingColumns(labels, getDataPointsOfVisibleSeries(this));

        if (areOverlapping) {
```

What would you think?

Comment (ptquang86) [reply src/components/visualizations/chart/highcharts/plugins/autohideLabels/autohideColumnLabels.ts]: updated in this commit https://github.com/gooddata/gooddata-react-components/pull/1106/commits/cf7d8cb98d0a0f19c1b165612e16891c480289d6

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4474/ : SUCCESS in 8m 24s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5748/ : SUCCESS in 12s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2401/ : SUCCESS in 13m 53s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4476/ : SUCCESS in 11m 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1132/ : SUCCESS in 27s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/1435/ : SUCCESS in 18m 45s

**Pull request merged**